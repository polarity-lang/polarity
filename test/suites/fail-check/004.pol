data Eq (a: Type, x y: a) {
    Refl(a: Type, x: a) : Eq(a, x, x)
}

codata Bool {
    (x: Bool).neg_inverse: Eq(Bool, x, x.not.not),
    and(other: Bool): Bool,
    not: Bool
}

comatch {
    neg_inverse => ?,
    and(other) => ?,
    not => ?
}: Bool
