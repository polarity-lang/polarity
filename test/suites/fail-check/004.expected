T-013

  × Local comatch not supported for type Bool because Bool contains destructors with self parameters
    ╭─[004.pol:12:1]
 12 │     def Unit.example : Bool {
 13 │ ╭─▶     Top => comatch {
 14 │ │                 .neg_inverse => ?,
 15 │ │                 .and(other) => ?,
 16 │ │                 .not => ?
 17 │ ╰─▶            }
 18 │     }
    ╰────
  help: Use a top-level codefinition instead
